plugins/inputs/fibaro/fibaro.go

```diff
@@ -174,6 +175,12 @@ func (f *Fibaro) Gather(acc telegraf.Accumulator) error {
 		}
 		fields := make(map[string]interface{})
 
+		if device.Properties.BatteryLevel != nil {
+			if fValue, err := strconv.ParseFloat(device.Properties.BatteryLevel.(string), 64); err == nil {
```

Comment:
Some post-merge review here: are we sure that `device.Properties.BatteryLevel` is always a string? this will panic if not. Also: if it's always a string, why not just type it as a string? We may be making more problems for ourselves by using interface{}

Reply:
The value is always quoted in the payload.
As said in my other comment, the JSON payload is dynamic and batteryLevel doesn't exist as a property for all devices but only for those which are battery operated.
How would you recommend to manage this case?

Reply:
You should still be able to decode a json object with missing keys into the struct. it'll just default to the zero value. eg https://play.golang.org/p/me0tDzHlQc_N

Reply:
OK but then if it defaults to zero, how do I differentiate zero from non existing in json payload? Because a not operated battery device doesn't have this info, it should not make it into the DB.

Reply:
In this case the "zero" value for a string is just an empty string, `""`, which you could still check for... but I see what you're getting at. If you don't want the fields reported when they're not there, you need some way to distinguish a nil from a zero value. 👍

Reply:
One way to handle this is to have the struct field be a pointer: `BatteryLevel *string`.  The value will be nil if not set in the document.

Reply:
Got it, thank you: https://github.com/influxdata/telegraf/pull/7411